### PR TITLE
PUB-1345 - Fix dot filenames

### PR DESCRIPTION
**Pull request by furnivall**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-1345

### Change description ###
Pre-fix:
The following error is thrown when any file with a dot before the file extension is uploaded. 

image

image

This also applies to the manual upload service: 

image

Problem is now fixed, hopefully.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/service/fileHandlingService.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class FileHandlingService {
   }
 
   isValidFileType(fileName: string, image: boolean): boolean {
-    const fileType = fileName.split('.')[1]?.toLocaleLowerCase();
+    const fileType = fileName.slice((fileName.lastIndexOf('.') - 1 >>> 0) + 2).toLocaleLowerCase();
```
Comment (kurt-uk) [inline src/main/service/fileHandlingService.ts:46]: would fileName.slice((fileName.lastIndexOf('.')) +1) not work and be less conversion? rather than converting to into which index already returns?

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/service/fileHandlingService.ts (hunk 1 above)
Comment (furnivall) [reply src/main/service/fileHandlingService.ts:46]: So - I ran a couple of tests on jsfiddle to check this. The complexity is there to handle the edge case where a file starts with a dot and has no extension (e.g. .gitignore or .htaccess). In those specific cases, the bit shift operator hack allows us to prevent it from returning anything (e.g. if you were to submit a file called ".png" or something to that effect, it would return an invalid filetype, whereas with the +1, it would allow that to be submitted).

It all comes down to what you're looking for, really.

**Review by kurt-uk — APPROVED** [review]

**Review by joshblackmoor — APPROVED** [review]
